test/js/style/style.test.js

```diff
@@ -162,6 +162,34 @@ test('Style#_broadcastLayers', function(t) {
     });
 });
 
+test('Style#_broadcastLayers with specific ids', function(t) {
+    var style = new Style({
+        'version': 8,
+        'sources': {
+            'source': {
+                'type': 'vector'
+            }
+        },
+        'layers': [
+            {id: 'first', source: 'source', type: 'fill', 'source-layer': 'source-layer'},
+            {id: 'second', source: 'source', type: 'fill', 'source-layer': 'source-layer'},
+            {id: 'third', source: 'source', type: 'fill', 'source-layer': 'source-layer'}
+        ]
+    });
+
+    style.on('error', function(error) { t.error(error); });
+
+    style.on('load', function() {
+        style.dispatcher.broadcast = function(key, value) {
+            t.equal(key, 'update layers');
+            t.deepEqual(value.map(function(layer) { return layer.id; }), ['second', 'third']);
+            t.end();
+        };
+
+        style._broadcastLayers(['second', 'third']);
```

Comment:
Agree with your points Lucas. My intent here was to introduce the minimum
amount of code changes to fix a severe regression and get it into a release
ASAP — most of the things you mentioned were there before the PR. Should we
merge now but follow up with another refactoring PR?

вторник, 23 февраля 2016 г. пользователь Lucas Wojciechowski написал:

> In test/js/style/style.test.js
> <https://github.com/mapbox/mapbox-gl-js/pull/2174#discussion_r53835329>:
>
> > +            {id: 'first', source: 'source', type: 'fill', 'source-layer': 'source-layer'},
> > +            {id: 'second', source: 'source', type: 'fill', 'source-layer': 'source-layer'},
> > +            {id: 'third', source: 'source', type: 'fill', 'source-layer': 'source-layer'}
> > +        ]
> > +    });
> > +
> > +    style.on('error', function(error) { t.error(error); });
> > +
> > +    style.on('load', function() {
> > +        style.dispatcher.broadcast = function(key, value) {
> > +            t.equal(key, 'update layers');
> > +            t.deepEqual(value.map(function(layer) { return layer.id; }), ['second', 'third']);
> > +            t.end();
> > +        };
> > +
> > +        style._broadcastLayers(['second', 'third']);
>
> Rather than testing the private method _broadcastLayers we should apply
> some style operations (like setLayoutProperty) that trigger
> _broadcastLayers. This will increase our test coverage #1992
> <https://github.com/mapbox/mapbox-gl-js/issues/1992> and make this test
> more robust against private method refactors.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mapbox/mapbox-gl-js/pull/2174/files#r53835329>.
>


-- 
Vladimir Agafonkin
http://agafonkin.com/en
+380 (93) 745 44 61